lib/gateway-kernel.coffee

```diff
@@ -0,0 +1,94 @@
+child_process = require 'child_process'
+path = require 'path'
+
+_ = require 'lodash'
+uuid = require 'uuid'
+services = require('./jupyter-js-services-shim')
+
+Kernel = require './kernel'
+
+class GatewayKernel extends Kernel
+    constructor: (kernelSpec, session) ->
+        super kernelSpec
+
+        @session = session
+        @session.statusChanged.connect(@_onStatusChange, this)
```

Comment:
On 21/07/16 14:40, Nikita Kitaev wrote:
> Apparently the second argument (the binding for this) is required by the signalling API being used. If |this| is not passed I'm pretty sure it will cause an error to be thrown.

This doesn't sound right.

In Javascript, if a callback requires `this` to be set by the caller. This is usually achieved by binding the function.

In Coffeescript, this is achieved by fat arrows (=>). Unbound functions are defined using (->).